IbisLib/gui/quadviewwindow.cpp

```diff
@@ -414,42 +406,54 @@ bool QuadViewWindow::eventFilter(QObject *obj, QEvent *event)
             }
         }
         if( which != -1 )
-            SetCurrentView( which );
+            SetCurrentViewWindow( which );
     }
     return false;
 }
 
-void QuadViewWindow::SetCurrentView( int index )
+void QuadViewWindow::SetExpandedView( bool on )
 {
-    if( m_sceneManager )
+    if( m_viewExpanded != on )
     {
-        switch(index)
+        ExpandViewButtonClicked();
+    }
+    ApplicationSettings * s = Application::GetInstance().GetSettings();
+    s->ExpandedView = m_viewExpanded;
+}
+
+void QuadViewWindow::SetCurrentViewWindow(int index )
+{
+    if( m_currentViewWindow == index )
+        return;
+    m_currentViewWindow = index;
+    for( int i = 0; i < 4; ++i )
+    {
+        if( i == m_currentViewWindow )
         {
-        case TRANSVERSE_WIN:
-            m_sceneManager->SetCurrentView( TRANSVERSE_VIEW_TYPE );
-            break;
-        case THREED_WIN:
-            m_sceneManager->SetCurrentView( THREED_VIEW_TYPE );
-            break;
-        case CORONAL_WIN:
-            m_sceneManager->SetCurrentView( CORONAL_VIEW_TYPE );
-            break;
-        case SAGITTAL_WIN:
-            m_sceneManager->SetCurrentView( SAGITTAL_VIEW_TYPE );
-            break;
+            m_vtkWindowFrames[i]->setStyleSheet( "background-color:red" );
         }
-        for( int i = 0; i < 4; ++i )
+        else
         {
-            if( i == index )
-            {
-                m_vtkWindowFrames[i]->setStyleSheet( "background-color:red" );
-            }
-            else
-            {
-                m_vtkWindowFrames[i]->setStyleSheet( "" );
-            }
+            m_vtkWindowFrames[i]->setStyleSheet( "" );
         }
     }
+    ApplicationSettings * s = Application::GetInstance().GetSettings();
```

Comment:
There should be a load/save settings in QuadView. Then you wouldn't need to do this.